File: sql.rb
require "erb"
require('csv')
csv = CSV.open('./bquxjob_56be86f2_183e85a52e1.csv').readlines

# Create template.
template = %q{
  WITH poap AS (
  SELECT *
  FROM (
  VALUES
  <%# ignore numerous minor requests -- focus on priorities %>
  % csv[1..-1].each_with_index do |c, i|
  %  id, poapId, name, address,createdAt,approvedAt,printedAt = c
  %  label = (name.match(/\.eth/) && name.split(".").length == 2) ?  "\'#{name.split(".").first}\'" : 'null'
  %  if poapId 
      (<%= poapId %>,'<%= createdAt %>','<%= address.downcase %>','<%= name %>', <%= label %> ,'<%= printedAt %>')<%= (i != csv.length - 2) ? ',' : '' %>
  %  end
  % end
  ) AS x (poap_id, created_at, address, name, label, printed_at)
  )
  select * from poap;
}.gsub(/^  /, '')

message = ERB.new(template, 0, "%<>")

# Produce result.
email = message.result
puts email

File: sql.js
const fs = require('fs').promises;
const namehash = require('@ensdomains/eth-ens-namehash')

async function main(){
    var csv = await fs.readFile("./data/data.csv", "utf8")
    csv = csv.split("\n")
    csv = csv.map((c) => c.split(","))    
    csv.shift()
    const doc = `
WITH ens_poapid_devcon6 AS (
SELECT *
FROM (
VALUES
${
    csv.map(c => {
        [id, poapId, name, address,createdAt,approvedAt,printedAt] = c  
        label = (name.match(/\.eth/) && name.split(".").length == 2) ?  `'${name.split(".")[0]}'` : 'null'
        node = namehash.hash(name)
        if(poapId){
            return `(${poapId}, '${createdAt}', '${address.toLowerCase()}','${name}', ${label}, '${node}', '${printedAt}')\n`
        }else{
            return null
        }
    }).filter(a => !!a)
}
) AS x (poap_id, created_at, address, name, label, node, printed_at)
)
select * from ens_poapid_devcon6;
`;
    
    console.log(doc)    
}
main()